sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClientBuilder.java

```diff
@@ -86,19 +83,12 @@ public BlobServiceClient buildClient() {
      * @return a {@link BlobServiceAsyncClient} created from the configurations in this builder.
      */
     public BlobServiceAsyncClient buildAsyncClient() {
+        BuilderHelper.validateCpk(customerProvidedKey, endpoint);
+
         BlobServiceVersion serviceVersion = version != null ? version : BlobServiceVersion.getLatest();
-        HttpPipeline pipeline = (httpPipeline != null) ? httpPipeline : BuilderHelper.buildPipeline(() -> {
-            if (storageSharedKeyCredential != null) {
-                return new StorageSharedKeyCredentialPolicy(storageSharedKeyCredential);
-            } else if (tokenCredential != null) {
-                return new BearerTokenAuthenticationPolicy(tokenCredential, String.format("%s/.default", endpoint));
-            } else if (sasTokenCredential != null) {
-                return new SasTokenCredentialPolicy(sasTokenCredential);
-            } else {
-                throw logger.logExceptionAsError(
-                    new IllegalArgumentException("Authorization credentials must be set."));
-            }
-        }, retryOptions, logOptions, httpClient, additionalPolicies, configuration, serviceVersion);
+        HttpPipeline pipeline = (httpPipeline != null) ? httpPipeline : BuilderHelper.buildPipeline(
```

Comment:
Could you add a check before this call to ensure there is a credential set in the builder? The changes made here no longer having it throw an `IllegalArgumentException` when no credential has been set.

Reply:
Good catch!